packages/driver/src/cypress/error_messages.coffee

```diff
@@ -1074,20 +1071,50 @@ module.exports = {
       URL from the `options` object: {{optionsUrl}}
       URL from the `url` parameter: {{url}}
     """
-    cannot_visit_2nd_domain: """
-      #{cmd('visit')} failed because you are attempting to visit a second unique domain.
+    cannot_visit_different_protocol: """
+      #{cmd('visit')} failed because you are attempting to visit a URL that is of a different origin.
+
+      The new URL is considered different origin because the 'protocol' is different.
+
+      You may only 'cy.visit()' same origin URLs within a single test.
+
+      The previous domain you visited was: '{{previousDomain}}'
```

Comment:
It prints the entire URL so that they can better compare what they visited. Having it just show `http` versus `https` I think would be not helpful - they have no idea what url caused the mishap.

Reply:
```suggestion
      The previous URL you visited was: '{{previousDomain}}'
```